README.md

```diff
@@ -497,13 +557,149 @@ New-AzureRmResourceGroup -Name "<name>" -Location "<location>"
 New-AzureRmResourceGroupDeployment -Name "<deployment name>" -ResourceGroupName "<name>" -TemplateUri "$templateBaseUrl/mainTemplate.json" -TemplateParameterObject $clusterParameters
 ```
 
-### Web based deploy
+## Configuring TLS
 
-<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Felastic%2Fazure-marketplace%2Fmaster%2Fsrc%2FmainTemplate.json" target="_blank">
-   <img alt="Deploy to Azure" src="http://azuredeploy.net/deploybutton.png"/>
-</a>
+It is strongly recommended that you secure communication when using the template
+in production. X-Pack Security can provide Authentication and Role Based Access control,
+and Transport Layer Security (TLS) can be configured for both Elasticsearch and Kibana.
+
+### TLS for Kibana
+
+You can secure external access from the browser to Kibana with TLS by supplying
+a certificate and private key in PEM format with `kibanaCertBlob` and
+`kibanaKeyBlob` parameters, respectively.
+
+### TLS for Elasticsearch Transport layer
+
+You can secure communication between nodes in the cluster with TLS on the
+Transport layer. Configuring TLS for the Transport layer requires
+`xPackPlugins` and `esTransportSecurity` be set to `Yes`. 
+
+You must supply a PKCS#12 archive with the `esTransportCaCertBlob` parameter (and optional
+passphrase with `esTransportCaCertPassword`) containing the CA which should be used to generate
+a certificate for each node within the cluster. An optional
+passphrase can be passed with `esTransportCertPassword` to encrypt the generated certificate
+on each node.
+
+A new CA PKCS#12 archive can be generated using [Elastic's certutil command](https://www.elastic.co/guide/en/elasticsearch/reference/current/certutil.html). The simplest command to generate a CA cert is
```

Comment:
The expectation here is that users will download Elasticsearch on their own and run `certutil`? It doesn't sound very streamlined experience, but I don't have any good alternative options to suggest.

Reply:
I'll make it clear that this is just a suggestion for how to create a CA cert, but that they may already have one, or could generate one in other ways e.g. through PowerShell